tests/ffuncs/test_ffunc_count.py

```diff
@@ -89,95 +104,168 @@ def test_single_arr_with_nan_workflow(self):
+    #    * includes NaN, but separate "validity" array
+    idx_with_empty_cell = iindex({(1,): [3]}, 0, (5,))  # [0, 0, 0, 1, 0]
+
+    dirty_weights = [9.0, 9.0, float("nan"), 9.0, 9.0]
+    clean_weights = [9.0, 9.0, MAXFLOAT, 9.0, 9.0]
+    weights_validity = [True, True, False, True, True]
+
+    args = [
+        dirty_weights,
+        (clean_weights, weights_validity),
+        (dirty_weights, weights_validity),
+    ]
+
+    @pytest.mark.parametrize("weights", args)
+    def test_propagate_missing_return_nan(self, weights):
+        WT = 1.0 if weights is None else 9.0
+        with compare_ccube_to_xcube():
+            # The cube of idx1 has rowids:
+            # 0           1
+            # [1, 3, 4]  [0, 2]
+            # and fact values:
+            # 0           1
+            # [2.0, 4.0, 5.0]  [1.0, nan]
+            counts = ccube([idx1]).count(weights)
+            # Cell (1,) MUST be missing, because it had a missing input.
+            assert arr_eq(counts, [3.0 * WT, float("nan")])
+
+            # The cube of idx1 x idx_with_empty_cell has fact values:
+            #      _______idx w_______
+            #      0           1
+            # i|0  [2.0, 5.0]  [4.0]
+            # d|
+            # x|
+            # 1|1  [1.0, nan]       []
+            counts = ccube([idx1, self.idx_with_empty_cell]).count(weights)
+            # Cell (1, 0) MUST be missing, because it had a missing input.
+            # Cell (1, 1) MUST be missing, because it had no inputs.
+            assert arr_eq(counts, [[2.0 * WT, 1.0 * WT], [float("nan"), float("nan")]])
+
+    @pytest.mark.parametrize("weights", args)
+    def test_ignore_missing_return_nan(self, weights):
+        WT = 1.0 if weights is None else 9.0
+        with compare_ccube_to_xcube():
+            # The cube of idx1 has rowids:
+            # 0           1
+            # [1, 3, 4]  [0, 2]
+            # and fact values:
+            # 0           1
+            # [2.0, 4.0, 5.0]  [1.0, nan]
+            counts = ccube([idx1]).count(weights, ignore_missing=True)
+            # Cell (1,) MUST NOT be missing, because it had a valid input.
+            assert arr_eq(counts, [3.0 * WT, 1.0 * WT])
+
+            # The cube of idx1 x idx_with_empty_cell has fact values:
+            #      _______idx w_______
+            #      0           1
+            # i|0  [2.0, 5.0]  [4.0]
+            # d|
+            # x|
+            # 1|1  [1.0, nan]       []
+            counts = ccube([idx1, self.idx_with_empty_cell]).count(
+                weights, ignore_missing=True
+            )
+            # Cell (1, 0) MUST NOT be missing, because it had a valid input.
+            # Cell (1, 1) MUST be missing, because it had no inputs.
+            assert arr_eq(counts, [[2.0 * WT, 1.0 * WT], [1.0 * WT, float("nan")]])
+
+    @pytest.mark.parametrize("weights", args)
+    def test_propagate_missing_return_validity(self, weights):
+        WT = 1.0 if weights is None else 9.0
```

Comment:
Basically everywhere where we're using that same `args` defined above. We need to either change the `args` to include `None` (if it still makes sense) or to fix all these lines to not hamper the readability. In that regard, we could also extract the `WT` outside, alongside `args`.